onfido-java/src/main/java/com/onfido/WebhookEventVerifier.java

```diff
@@ -0,0 +1,42 @@
+package com.onfido;
+
+import com.onfido.api.ApiJson;
+import com.onfido.api.WebhookEvent;
+import com.onfido.api.WebhookPayload;
+import com.onfido.exceptions.OnfidoException;
+import org.apache.commons.codec.binary.Hex;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+
+public class WebhookEventVerifier {
+
+    private ApiJson<WebhookPayload> parser = new ApiJson<>(WebhookPayload.class);
+    private final String webhookToken;
+
+
+    public WebhookEventVerifier(String webhookToken) {
+        this.webhookToken = webhookToken;
+    }
+
+    public WebhookEvent readPayload(String rawEventBody, String hexSignature) throws OnfidoException {
+        try {
```

Comment:
I think it will lead to better error messages if you wrap the individual parts that throw exceptions in their own try/catch

i.e. the `sha256Hmac.init(secretKey);` line can catch and throw `new OnfidoException("Invalid webhook token", e);`, and you can just throw a `new OnfidoException("Invalid signature for webhook event")` directly if the signatures aren't equal.